baseapp/baseapp.go

```diff
@@ -810,9 +809,8 @@ func (app *BaseApp) DeliverTx(req abci.RequestDeliverTx) (res abci.ResponseDeliv
 		msg := tx.GetMsg()
 		messageType = msg.Type()
 		recipient = msg.GetRecipient()
-		if signerPK == nil || messageType == types2.MsgAppStakeName {
-			signers := msg.GetSigners()
-			if len(signers) >= 1 {
+		if signerPK == nil {
```

Comment:
Unfortunately no.  This was introduced by the large, combined commit  32157db43c88fcd37d5c68978f0256c8631db833 of v0.8, but this part does nothing meaningful.  MsgAppStake's `GetSigners` always returns a single element array and its value is `msg.PubKey` which was guaranteed to be the signer, so it doesn't make any sense to have this.

Reply:
Logically, I agree that you're right and I cannot prove otherwise. However, I'm really paranoid that there's some weird edge case and I don't want any of us to deal with a chain halt.

`Hypothesis`: When the message type is NOT an AppStake (e.g. delegated staking for suppliers / node providers), maybe we need to enforce to using the first signer?

Reply:
This patch does not change the logic in the case where the message type is not an AppStake.  Regardless of this patch, we enforce to using the first signer only if `signerPK` is null.  Moreover, this `signer` affects `abci.ResponseDeliverTx` which just goes to the transaction log.  I don't think it causes a chain halt.